### PR TITLE
orderbook_serialization_issue

### PR DESCRIPTION
**Pull request by koloale**

### CODE REVIEW

File: src/main/scala/scorex/transaction/assets/exchange/Order.scala [hunk 1]
```diff
@@ -167,7 +166,7 @@ case class Order(@ApiModelProperty(dataType = "java.lang.String") senderPublicKe
     }
   }
```
Comment (ismagin) [inline src/main/scala/scorex/transaction/assets/exchange/Order.scala]: please fix equals too

File: src/main/scala/com/wavesplatform/matcher/model/EventSerializers.scala [hunk 2]
```diff
@@ -76,14 +76,14 @@ object MatcherSerializer {
...
-      JsSuccess(TreeMap.empty[Price, Level[BuyLimitOrder]] ++ a)
+      JsSuccess(TreeMap.empty[Price, Level[BuyLimitOrder]](OrderBook.bidsOrdering) ++ a)
     }
   }
 
   implicit val sellOrderTreeMapReads = new Reads[TreeMap[Price, Level[SellLimitOrder]]] {
     override def reads(jv: JsValue): JsResult[TreeMap[Price, Level[SellLimitOrder]]] = {
       val a = jv.as[Map[String, Level[SellLimitOrder]]].map { case(k, v) => (k.toLong, v) }
-      JsSuccess(TreeMap.empty[Price, Level[SellLimitOrder]] ++ a)
+      JsSuccess(TreeMap.empty[Price, Level[SellLimitOrder]](OrderBook.asksOrdering) ++ a)
```
Comment (phearnot) [inline src/main/scala/com/wavesplatform/matcher/model/EventSerializers.scala]: TreeMap is serialized as a JSON object, which is [an unordered set of name/value pairs](http://www.json.org/). Explicitly specifying ordering isn't going to fix an issue. If you need to guarantee level ordering, you probably need to serialize bids/asks as an array.

Comment (phearnot) [inline src/main/scala/com/wavesplatform/matcher/model/EventSerializers.scala]: Disregard that.

File: src/main/scala/scorex/transaction/assets/exchange/Order.scala (hunk 1 above)
Comment (koloale) [inline src/main/scala/scorex/transaction/assets/exchange/Order.scala]: What's wrong with equals? Currently, fields, which are used in equals also used to create idStr hence hashcode

**Pull request merged**